### PR TITLE
Increase `allowedExecutionNameLength` to 63

### PR DESCRIPTION
**Pull request by rahul-theorem**
Signed-off-by: Rahul Mehta <98349643+rahul-theorem@users.noreply.github.com>

# TL;DR
Increase `allowedExecutionNameLength` from 20 to 63 to allow more legible human-readable execution IDs

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
See title/summary

## Tracking Issue
Closes https://github.com/flyteorg/flyte/issues/2824

## Follow-up issue
_NA_

### CODE REVIEW

File: pkg/manager/impl/validation/execution_validator.go
Comment (katrogan) [inline pkg/manager/impl/validation/execution_validator.go:20]: should we also update the name length for [generated names](https://github.com/flyteorg/flyteadmin/blob/master/pkg/common/executions.go#L11)?

Comment (rahul-theorem) [reply pkg/manager/impl/validation/execution_validator.go:20]: Defer to other maintainers, seems OK to leave the randomly generated string short (since a longer name might make the frontend less legible). If it's preferable to have both be consistent, I'm happy to make the change (cc @kumare3 @eapolinario)

Comment (katrogan) [reply pkg/manager/impl/validation/execution_validator.go:20]: sure, just wanted to check if the omission was intentional :)
I think the chance of collisions for the 20 char generated name isn't a concern, so keeping as is sounds good 👍

**Comment by rahul-theorem** [comment]
Just fixed the failing test, @katrogan would it be possible to re-run the CI checks (looks like it's required since this is my first PR to flyteadmin)?

**Review by katrogan — APPROVED** [review]

**Pull request merged**
